tests/Unit/Elliptic/Systems/Xcts/Events/Test_ObserveAdmIntegrals.cpp

```diff
@@ -0,0 +1,163 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Determinant.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Slice.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/AreaElement.hpp"
+#include "Domain/CoordinateMaps/Composition.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.tpp"
+#include "Domain/CoordinateMaps/Wedge.hpp"
+#include "Domain/CreateInitialElement.hpp"
+#include "Domain/Creators/Sphere.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/ElementToBlockLogicalMap.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Domain/InterfaceLogicalCoordinates.hpp"
+#include "Domain/Structure/IndexToSliceAt.hpp"
+#include "Domain/Structure/InitialElementIds.hpp"
+#include "Elliptic/Systems/Xcts/Events/ObserveAdmIntegrals.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/ProjectToBoundary.hpp"
+#include "NumericalAlgorithms/LinearOperators/DefiniteIntegral.hpp"
+#include "NumericalAlgorithms/Spectral/LogicalCoordinates.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrHorizon.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Xcts/WrappedGr.hpp"
+
+namespace {
+
+using KerrSchild = Xcts::Solutions::WrappedGr<gr::Solutions::KerrSchild>;
+
+void test_compute_linear_momentum(const double distance,
+                                  const size_t refinement_level,
+                                  const size_t polynomial_order) {
+  // Define black hole parameters
+  const double mass = 1;
+  const double boost_speed = 0.5;
+  const double lorentz_factor = 1. / sqrt(1. - square(boost_speed));
+
+  // Get Kerr-Schild solution
+  const std::array<double, 3> dimensionless_spin{{0., 0., 0.}};
+  const std::array<double, 3> boost_velocity{{0., 0., boost_speed}};
+  const KerrSchild solution{mass, dimensionless_spin,
```

Comment:
Please also test this with a nontrivial conformal factor. `WrappedGr` sets $\psi=1$ (see its docs), so any dependence on the conformal decomposition is not yet tested here and the powers of the conformal factor are very easy to get wrong.

One way to test this is by using a Schwarzschild solution in, e.g., isotropic coordinates (see `src/PointwiseFunctions/AnalyticSolutions/Xcts/Schwarzschild.hpp`) and apply a Lorentz boost to all quantities (see e.g. `test_boosted_spacetime_metric` in `tests/Unit/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Test_KerrSchild.cpp`).

Reply:
I believe it is a good idea to work on this test in a separate PR. I've added my partial implementation in this branch: https://github.com/iago-mendes/spectre/tree/boosted-schwarzschild